src/compiler.ts

```diff
@@ -6185,6 +6190,7 @@ export class Compiler extends DiagnosticEmitter {
         valueExpr = this.makeReplace(
           valueExpr,
           module.local_get(localIndex, type.toNativeType()),
+          type,
```

Comment:
Probably not a big deal the way it is, but `type` and `valueType` are ambiguous here in that `makeReplace` can do a `__release` with `type` and a `__retain` with `valueType`, technically. Now we know these must be generally compatible, but still either of these might be nullable or not nullable, which might at some point lead to subtle differences down the road. Suggesting to add two types `newType` and `oldType` to `makeReplace` for clarity, and to avoid that we run into problems in the future. In this case that'd be `oldType=type` and `newType=valueType`.